framework/auth/views.py

```diff
@@ -637,3 +644,116 @@ def resend_confirmation_post(auth):
 
     # Don't go anywhere
     return {'form': form}
+
+
+def external_login_email_get():
+    """
+    Landing view for first-time oauth-login user to enter their email address.
+    HTTP Method: GET
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_external_first_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    external_id_provider = session.data['auth_user_external_id_provider']
+
+    return {
+        'form': form,
+        'external_id_provider': external_id_provider.upper()
+    }
+
+
+def external_login_email_post():
+    """
+    View to handle email submission for first-time oauth-login user.
+    HTTP Method: POST
+    """
+
+    form = ResendConfirmationForm(request.form)
+    session = get_session()
+    if not session.is_external_first_login:
+        raise HTTPError(http.UNAUTHORIZED)
+
+    external_id_provider = session.data['auth_user_external_id_provider']
+    external_id = session.data['auth_user_external_id']
+    fullname = session.data['auth_user_fullname']
+
+    if form.validate():
+        clean_email = form.email.data
+        user = get_user(email=clean_email)
+        external_identity = {
+            external_id_provider: {
+                'id': external_id,
+                'status': None,
+            }
+        }
+        if user:
+            # 0. check if this user is already linked with other profile
+            # 1. update user oauth
+            # 2. send confirmation email
+            # 3. notify user
+            # 4. remove session and osf cookie
+            external_status = ''
+            if user.external_identity:
+                if external_id_provider in user.external_identity:
+                    if user.external_identity[external_id_provider]:
+                        external_status = user.external_identity[external_id_provider]['status']
+
+            # TODO: handle pending status: the current or another user also claimed this account but not confirmed
```

Comment:
I think this would only be hit if the current user had already claimed, but if so shouldn't this flow be avoided (because `get_user_from_cas_resp` would return `user, 'authenticate'` instead of `None, 'oauth_first_login'`)?